WorkAndFun/src/test/java/pl/infoshare/workandfun/announcements/AnnouncementServiceTest.java

```diff
@@ -0,0 +1,141 @@
+        assertThat(firstElement.getId()).isEqualTo(1L);
+        assertThat(firstElement.getPrice()).isEqualTo("200");
+        assertThat(firstElement.getIsPriceNegotiable()).isFalse();
+        assertThat(firstElement.getFullLocalization()).isEqualTo("Warszawa, dzielnica, osiedle");
+        assertThat(firstElement.isIndividualPrice()).isFalse();
+    }
+
+    @Test
+    void shouldFindAllByQuerySpec() {
+        announcementService.findAllByQuerySpec(mock(AnnouncementSpec.class));
+        verify(announcementsRepository).findAll(any(AnnouncementSpec.class));
+    }
+
+    @Test
+    void shouldThrowExceptionWhenFindByIdNotFoundAnnouncement() {
+        //given
+        Long id3 = 3L;
+        when(announcementsRepository.findById(id3)).thenReturn(Optional.empty());
+        //when
+        //then
+        Assertions.assertThrows(AnnouncementNotFoundException.class, () -> announcementService.findById(id3));
+    }
+
+    @Test
+    void shouldFindById() {
+        //given
+        when(announcementsRepository.findById(id)).thenReturn(Optional.of(announcement));
+        //when
+        //then
+        assertThat(announcement).isEqualTo(announcementService.findById(id));
+    }
+
+    @Test
+    void shouldFindByIdAndMapToDto() {
+        //given
+        when(announcementsRepository.findById(id)).thenReturn(Optional.of(announcement));
+        //when
+        final var result = announcementService.findByIdConvertToDto(id);
+        //then
+        assertThat(result.getId()).isEqualTo(1L);
+        assertThat(result.getType()).isEqualTo(Type.SERVICE_OFFER);
+        assertThat(result.getHeader()).isEqualTo("Wyprowadzam psy, koty, myszy, konie, słonie");
+        assertThat(result.getServiceType()).isEqualTo(ServiceType.INNE);
+        assertThat(result.getCity()).isEqualTo("Warszawa");
+        assertThat(result.getCityDistrict()).isEqualTo("dzielnica");
+        assertThat(result.getUnit()).isEqualTo("osiedle");
+        assertThat(result.getPrice()).isEqualTo("200");
+        assertThat(result.getVoivodeship()).isEqualTo(Voivodeship.MAZOWIECKIE);
+        assertThat(result.getEmail()).isEqualTo("andrzej@aa.pl");
+        assertThat(result.getIsPriceNegotiable()).isFalse();
+        assertThat(result.getDescription()).isEqualTo("Andrzej, czyli ja to miłośnik zwierząt chętnie spędzający z nimi czas, nie masz co zrobić ze swoim zwierzakiem, zadzwoń do Andrzeja");
+        assertThat(result.getPhoneNumber()).isEqualTo("+48666666666");
+        assertThat(result.getPriceAdditionComment()).isEqualTo("z FV będzie drożej");
+    }
+
+    @Test
+    void shouldThrowExceptionWhenFindByIdConvertToDtoNotFoundAnnouncement() {
+        //given
+        when(announcementsRepository.findById(id)).thenReturn(Optional.empty());
+        //when
+        //then
+        Assertions.assertThrows(AnnouncementNotFoundException.class, () -> announcementService.findById(id));
```

Comment:
```suggestion
        Assertions.assertThrows(AnnouncementNotFoundException.class, () -> announcementService.findByIdConvertToDto(id));
```
Jest błąd, bo testujesz inną metodę klasy AnnouncementService. Samo findById() zostało przetestowane wcześniej.
Poza tym proponuję bazować się na moich sugestiach, co pisałem wcześniej, przy sprawdzeniu wyjątku od findById()